pkg/operator/ceph/controller/spec.go

```diff
@@ -825,6 +825,37 @@ func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Contai
 	}
 }
 
+// rgw operations will be logged in sidecar ops-log
+func RgwOpsLogContainer(daemonID, ns string, c cephv1.ClusterSpec, Resources v1.ResourceRequirements) *v1.Container {
+
+	// Generate ops log file name based on daemon ID
+	opsLogFile := fmt.Sprintf("ops-log-%s", daemonID)
+	cmdLogRotate := "true"
+
+	if c.LogCollector.Enabled {
+		maxLogSize, periodicity := GetLogRotateConfig(c)
+		cmdLogRotate = fmt.Sprintf(cronLogRotate, opsLogFile, periodicity, maxLogSize.String(), "7")
+	}
+
+	return &v1.Container{
+		Name: "ops-log",
+		Command: []string{
+			"bash",
+			"-x", // Enable debugging mode
+			"-m", // Enable job control
+			"-c", // Run the following command
+			fmt.Sprintf("(%s) & tail -n+1 -F %s", cmdLogRotate, path.Join(config.VarLogCephDir, opsLogFile+".log")),
```

Comment:
Let me make sure I understand this:
1. Run the logrotate command in the background
2. tail the log

Is that right?

If so, we should not be doing this. Logrotate should be its own sidecar so that the process can be restarted by the container runtime if the process fails or stops for any reason.

Reply:
@parth-gr do you know if the Ceph log collector can collect these logs? If so, it might be most appropriate to integrate this feature with log collector for rotation.